File: tasks/day-2/golang/main.go
package main

import (
    "fmt"
    "os"
    "io/ioutil"
    "./present"
    "strings"
    "strconv"
)

func main() {
    var total_square_feets, ribon_length int

    file_path := os.Args[1]
    content, _ := ioutil.ReadFile(file_path)
    dimensions := strings.Split(string(content), "\n")

    for _, dimension := range dimensions {
        size := strings.Split(string(dimension), "x")
        width, _ := strconv.Atoi(size[0])
        length, _ := strconv.Atoi(size[1])
        height, _ := strconv.Atoi(size[2])
        pres := present.Present{width, length, height}
        ribon_length += pres.GetRibonLenght()
        total_square_feets += pres.GetSquare()
    }

    fmt.Printf("total square feets: %d\nRibon length: %d\n", total_square_feets, ribon_length)
}

File: tasks/day-2/python/main.py
import sys, present

file_path = sys.argv[1]
total_square_feets, ribon_length = 0, 0

with open(file_path, 'r') as presents:
    for present_params in presents:
        length, width, height = map(int, present_params.split('x'))
        p = present.Present(length, width, height)
        
        total_square_feets += p.get_square()
        ribon_length += p.get_ribon_lenght()

print('Total square feets: {}\nRibon length: {}'.format(total_square_feets, ribon_length))